src/common/mod.rs

```diff
@@ -97,6 +117,7 @@ impl<T: Clone> SenderWithContext<T> {
         }
     }
 
+    /// Updates this [`SenderWithContext`]'s [`ErrorContext`].
```

Comment:
Maybe add something like "this is the way one adds a call to the error context and works this way in order not to allocate on the heap"?